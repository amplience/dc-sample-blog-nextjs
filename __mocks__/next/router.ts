export const mockRouterOn = jest.fn();
export const Router = { events: { on: mockRouterOn } };
export const useRouter = (): { asPath: string; query: unknown } => ({ asPath: 'mocked-path', query: {} });
