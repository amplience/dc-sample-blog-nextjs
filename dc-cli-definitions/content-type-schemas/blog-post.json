{
  "body": "../../schemas/blog-post.json",
  "schemaId": "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/blog-post.json",
  "validationLevel": "CONTENT_TYPE"
}