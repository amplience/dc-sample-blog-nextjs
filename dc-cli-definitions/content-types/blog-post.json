{
  "contentTypeUri": "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/blog-post.json",
  "status": "ACTIVE",
  "settings": {
    "label": "Blog post",
    "icons": [
      {
        "size": 256,
        "url": "https://i1.adis.ws/i/ampproduct/icon-blogarticle"
      }
    ],
    "visualizations": [
      {
        "label": "Visualisation",
        "templatedUri": "https://blog.example.com/visualization.html?vse={{vse.domain}}&content={{content.sys.id}}",
        "default": true
      }
    ],
    "cards": [
      {
        "templatedUri": "https://bigcontent.io/cms/cards/summary-photo/index.html?headline=/title&image=/image/image&imageAlt=/image/altText&src=%2F%2F{{vse.domain}}%2Fcms%2Fcontent%2Fquery%3FfullBodyObject%3Dtrue%26query%3D%257B%2522sys.iri%2522%253A%2522http%253A%252F%252Fcontent.cms.amplience.com%252F{{content.sys.id}}%2522%257D%26scope%3Dtree%26store%3Dstaging"
      }
    ]
  },
  "repositories": ["content"]
}