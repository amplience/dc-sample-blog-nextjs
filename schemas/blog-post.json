{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "id": "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/blog-post.json",
  "title": "Blog post",
  "description": "A blog post",
  "allOf": [
    {
      "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content"
    }
  ],
  "type": "object",
  "properties": {
    "title": {
      "title": "Title",
      "description": "Used for heading and SEO title tag",
      "type": "string",
      "minLength": 1,
      "maxLength": 150
    },
    "_meta":{
      "type":"object",
      "properties":{
        "deliveryKey":{
          "type":"string",
          "title":"Delivery key",
          "description": "The delivery key is used as the URL slug"
        }
      },
      "required": ["deliveryKey"]
    }, 
    "date": {
      "title": "Creation date",
      "description": "Creation date (YYYY-MM-DD)",
      "type": "string",
      "maxLength": 10,
      "minLength": 10,
      "pattern": "^[0-9]{4}-((0[1-9])|(1[0-2]))-((0[1-9])|([1-2][0-9])|(3[0-1]))$"
    },
    "description": {
      "title": "Description",
      "description": "Used for blog listing page and SEO description",
      "type": "string",
      "minLength": 1,
      "maxLength": 200
    },
    "image": {
      "title": "Image",
      "description": "Used for the blog post's thumbnail and banner",
      "allOf": [
        {
          "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content-link"
        },
        {
          "properties": {
            "contentType": {
              "enum": ["https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/image.json"]
            }
          }
        }
      ]
    },
    "tags": {
      "title": "Tags",
      "description": "Blog tags",
      "type": "array",
      "minItems": 0,
      "maxItems": 10,
      "items": {
        "type": "string",
        "minLength": 1,
        "maxLength": 100,
        "title": "Tag",
        "description": ""
      }
    },
    "readTime": {
      "title": "Read time",
      "description": "The time it takes to read the blog",
      "type": "integer"
    },
    "authors": {
      "title": "Blog author",
      "description": "Article author(s) - max 3",
      "type": "array",
      "minItems": 1,
      "maxItems": 3,
      "items": {
        "allOf": [
          {
            "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content-link"
          },
          {
            "properties": {
              "contentType": {
                "enum": ["https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/author.json"]
              }
            }
          }
        ]
      }
    },
    "content": {
      "title": "Content",
      "description": "",
      "type": "array",
      "minItems": 1,
      "maxItems": 20,
      "items": {
        "allOf": [
          {
            "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content-link"
          },
          {
            "properties": {
              "contentType": {
                "title": "Content",
                "enum": [
                  "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/image.json",
                  "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/video.json",
                  "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/text.json"
                ]
              }
            }
          }
        ]
      }
    }
  },
  "propertyOrder": ["title", "_meta", "authors", "date", "description", "image", "tags", "readTime", "content"],
  "required": ["title", "authors", "date", "description", "image", "readTime", "content"]
}
