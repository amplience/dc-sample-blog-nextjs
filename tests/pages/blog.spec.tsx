/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import BlogPostPage from '../../pages/blog/[...slug]';
import blogPostFixture from '../fixtures/single-blog-post-data-object.json';
import { getBlogPostByDeliveryKey } from '../../common/services/blog-post.service';
import { NextPageContext } from 'next';

jest.mock('../../common/services/blog-post.service', () => {
  return {
    __esModule: true,
    getBlogPostByDeliveryKey: jest.fn()
  };
});

const mockGetBlogPostByDeliveryKey: jest.MockedFunction<typeof getBlogPostByDeliveryKey> = getBlogPostByDeliveryKey as jest.MockedFunction<
  typeof getBlogPostByDeliveryKey
>;

describe('Blog page', () => {
  let blogPost;

  beforeEach(() => {
    process.env.DYNAMIC_CONTENT_REFERENCE_ID = 'reference-id';
    process.env.GA_TRACKING_ID = 'ga-tracking-id';
    process.env.TAGS_FACET_FIELD = 'tags';
    blogPost = { ...blogPostFixture };
    blogPost._meta.toJSON = () => ({});
    jest.clearAllMocks();
  });

  test('renders Blog page with content', () => {
    const component = renderer.create(<BlogPostPage blogPost={blogPost} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('getInitialProps returns blog post content via api', async () => {
    mockGetBlogPostByDeliveryKey.mockResolvedValue(blogPost);
    const query = { slug: ['blog-slug'] };
    const result = await BlogPostPage.getInitialProps(({ query } as unknown) as NextPageContext);
    expect(result).toEqual({ blogPost: blogPostFixture });
  });

  test('getInitialProps should call getHydratedBlogList with staging environment', async () => {
    mockGetBlogPostByDeliveryKey.mockResolvedValue(blogPost);
    const query = { vse: 'vse-base-url', slug: ['blog-slug'] };
    await BlogPostPage.getInitialProps(({ query } as unknown) as NextPageContext);
    expect(mockGetBlogPostByDeliveryKey).toHaveBeenCalledWith('blog-slug', `//${query.vse}`);
  });

  test('getInitialProps should call getHydratedBlogList without staging environment', async () => {
    mockGetBlogPostByDeliveryKey.mockResolvedValue(blogPost);
    const query = { slug: ['blog-slug'] };
    await BlogPostPage.getInitialProps(({ query } as unknown) as NextPageContext);
    expect(mockGetBlogPostByDeliveryKey).toHaveBeenCalledWith('blog-slug', undefined);
  });

  test('getInitialProps throws error when deliveryKey is undefined', async () => {
    mockGetBlogPostByDeliveryKey.mockRejectedValue(new Error());
    const query = {};
    await expect(BlogPostPage.getInitialProps(({ query } as unknown) as NextPageContext)).rejects.toThrowError(Error);
  });
});
