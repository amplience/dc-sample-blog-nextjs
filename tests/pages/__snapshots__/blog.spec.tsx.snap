// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog page renders Blog page with content 1`] = `
<div
  className="jsx-3319508426 site-container"
>
  <section
    className="jsx-2398603546"
  >
    <header
      className="jsx-2398603546"
    >
      <a
        aria-label=""
        className="jsx-3354872520"
        href="/"
      >
        <img
          alt="Amplience logo"
          className="jsx-2718936118 logo"
          src="/static/images/logo-white-100h.png"
          srcSet="/static/images/logo-white-50h.png 1x, /static/images/logo-white-100h.png 2x"
        />
      </a>
      <span
        className="jsx-2398603546"
      >
        PRODUCT BLOG
      </span>
    </header>
  </section>
  <div
    className="jsx-3319508426 site-content"
  >
    <main
      className="jsx-3319508426"
    >
      <div
        className="jsx-452798245 content-header"
      >
        <section
          className="jsx-3378232041"
        >
          <div
            className="jsx-3378232041 avatars"
          >
            <div
              className="jsx-2854206183"
            >
              <div
                className="jsx-2854206183 avatar"
              >
                <picture>
                  <source
                    srcSet="https://i1.adis.ws/i/blogltd/man-business-suit?&w=36&h=36&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 1x, https://i1.adis.ws/i/blogltd/man-business-suit?&w=72&h=72&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 2x"
                    type="image/webp"
                  />
                  <source
                    srcSet="https://i1.adis.ws/i/blogltd/man-business-suit?&w=36&h=36&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 1x, https://i1.adis.ws/i/blogltd/man-business-suit?&w=72&h=72&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 2x"
                  />
                  <img
                    alt="The dashing Sir Blog avatar"
                    src="https://i1.adis.ws/i/blogltd/man-business-suit?&w=36&h=36&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}"
                  />
                </picture>
              </div>
            </div>
          </div>
          <div
            className="jsx-3378232041 publish-data"
          >
            <div
              className="jsx-3378232041 authors"
            >
              <div
                className="jsx-3680733511 name"
              >
                Sir Blog
              </div>
            </div>
            <div
              className="jsx-3378232041 publish-metadata"
            >
              <div
                className="jsx-3378232041 date"
              >
                August 07, 2019
              </div>
              <div
                className="jsx-3378232041 readTime"
              >
                12
                 mins read
              </div>
            </div>
          </div>
        </section>
        <div
          className="jsx-1065517858 tag-chips"
        >
          <div
            className="jsx-1065517858"
            onClick={[Function]}
          >
            sir blog
          </div>
          <div
            className="jsx-1065517858"
            onClick={[Function]}
          >
            you're it
          </div>
          <div
            className="jsx-1065517858"
            onClick={[Function]}
          >
            No tiggy backs
          </div>
        </div>
        <section
          className="jsx-3528769338"
        >
          <h1
            className="jsx-641291359"
          >
            My First Blog Post
          </h1>
          <h2
            className="jsx-1054465095"
          >
            Some text for the description that explains this blog post
          </h2>
        </section>
      </div>
      <div
        className="jsx-452798245 blog-image"
      >
        <picture>
          <source
            media="(min-width: 1080px)"
            srcSet="https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=2048&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 1x, https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=4096&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 2x"
            type="image/webp"
          />
          <source
            media="(min-width: 768px)"
            srcSet="https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=1080&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 1x, https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=2160&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 2x"
            type="image/webp"
          />
          <source
            media="(min-width: 480px)"
            srcSet="https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=768&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 1x, https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=1536&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 2x"
            type="image/webp"
          />
          <source
            srcSet="https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=480&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 1x, https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=960&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}&fmt=webp 2x"
            type="image/webp"
          />
          <source
            media="(min-width: 1080px)"
            srcSet="https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=2048&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 1x, https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=4096&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 2x"
          />
          <source
            media="(min-width: 768px)"
            srcSet="https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=1080&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 1x, https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=2160&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 2x"
          />
          <source
            media="(min-width: 480px)"
            srcSet="https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=768&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 1x, https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=1536&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 2x"
          />
          <source
            srcSet="https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=480&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 1x, https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=960&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 2x"
          />
          <img
            alt="crotch bags"
            src="https://i1.adis.ws/i/blogltd/friends-shopping-bags?&w=480&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h}"
          />
        </picture>
      </div>
      <div
        className="jsx-452798245 content-wrapper"
      >
        <section
          className="jsx-1163017566"
        >
          <div
            className="jsx-1163017566 content-placeholder"
          />
          <div
            className="jsx-1163017566"
          >
            <h1
              className="jsx-2595089998"
            >
              This is my first Blog
            </h1>
            <h2
              className="jsx-2595089998"
            >
              Look-a-header 2
            </h2>
            <h3
              className="jsx-2595089998"
            >
              Look-a-header 3
            </h3>
            <h4
              className="jsx-2595089998"
            >
              Look-a-header 4
            </h4>
            <h5
              className="jsx-2595089998"
            >
              Look-a-header 5
            </h5>
            <h6
              className="jsx-2595089998"
            >
              Look-a-header 6
            </h6>
            <p
              className="jsx-430790654"
            >
              <em>
                <strong>
                  This
                </strong>
              </em>
               
              <em>
                is
              </em>
               
              <em>
                some
              </em>
               
              <del>
                sample
              </del>
               
              <strong>
                text
              </strong>
              .
            </p>
            <p
              className="jsx-430790654"
            >
              <img
                alt="alt text"
                src="https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png"
                title="Logo Title Text 1"
              />
            </p>
            <p
              className="jsx-430790654"
            >
              This is some more sample text with a 
              <span
                className="jsx-3731109703"
              >
                <a
                  className="jsx-3655793200"
                  href="http://bbc.co.uk"
                  rel="noreferrer"
                  target="_new"
                >
                  LINK
                </a>
              </span>
            </p>
            <blockquote
              className="jsx-2064009043"
            >
              <p
                className="jsx-430790654"
              >
                Comments are cool!
              </p>
              <p
                className="jsx-430790654"
              >
                See what I mean
              </p>
            </blockquote>
            <p
              className="jsx-430790654"
            >
              Look some code:
            </p>
            <pre
              style={
                Object {
                  "MozHyphens": "none",
                  "MozTabSize": "4",
                  "OTabSize": "4",
                  "WebkitHyphens": "none",
                  "background": "#1d1f21",
                  "borderRadius": "0.3em",
                  "color": "#c5c8c6",
                  "direction": "ltr",
                  "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
                  "hyphens": "none",
                  "lineHeight": "1.5",
                  "margin": ".5em 0",
                  "msHyphens": "none",
                  "overflow": "auto",
                  "padding": "1em",
                  "tabSize": "4",
                  "textAlign": "left",
                  "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
                  "whiteSpace": "pre",
                  "wordBreak": "normal",
                  "wordSpacing": "normal",
                }
              }
            >
              <code
                style={
                  Object {
                    "MozHyphens": "none",
                    "MozTabSize": "4",
                    "OTabSize": "4",
                    "WebkitHyphens": "none",
                    "color": "#c5c8c6",
                    "direction": "ltr",
                    "float": "left",
                    "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
                    "hyphens": "none",
                    "lineHeight": "1.5",
                    "msHyphens": "none",
                    "paddingRight": "10px",
                    "tabSize": "4",
                    "textAlign": "left",
                    "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
                    "whiteSpace": "pre",
                    "wordBreak": "normal",
                    "wordSpacing": "normal",
                  }
                }
              >
                <span
                  className="react-syntax-highlighter-line-number"
                >
                  1

                </span>
                <span
                  className="react-syntax-highlighter-line-number"
                >
                  2

                </span>
              </code>
              <code
                style={
                  Object {
                    "MozHyphens": "none",
                    "MozTabSize": "4",
                    "OTabSize": "4",
                    "WebkitHyphens": "none",
                    "color": "#c5c8c6",
                    "direction": "ltr",
                    "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
                    "hyphens": "none",
                    "lineHeight": "1.5",
                    "msHyphens": "none",
                    "tabSize": "4",
                    "textAlign": "left",
                    "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
                    "whiteSpace": "pre",
                    "wordBreak": "normal",
                    "wordSpacing": "normal",
                  }
                }
              >
                <span
                  style={Object {}}
                >
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#96CBFE",
                      }
                    }
                  >
                    const
                  </span>
                  <span
                    style={Object {}}
                  >
                     something 
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#EDEDED",
                      }
                    }
                  >
                    =
                  </span>
                  <span
                    style={Object {}}
                  >
                     
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#A8FF60",
                      }
                    }
                  >
                    'anything'
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#c5c8c6",
                      }
                    }
                  >
                    ;
                  </span>
                  <span
                    style={Object {}}
                  >
                    

                  </span>
                </span>
                <span
                  style={Object {}}
                >
                  <span
                    style={Object {}}
                  >
                    
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#DAD085",
                      }
                    }
                  >
                    calls
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#c5c8c6",
                      }
                    }
                  >
                    (
                  </span>
                  <span
                    style={Object {}}
                  >
                    something
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#c5c8c6",
                      }
                    }
                  >
                    )
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#c5c8c6",
                      }
                    }
                  >
                    ;
                  </span>
                </span>
              </code>
            </pre>
            <p
              className="jsx-430790654"
            >
              inline 
              <code
                className="jsx-3800523359"
              >
                code
              </code>
               is awesome underlined
            </p>
            <hr />
            &lt;h4&gt;A title&lt;/h4&gt;
            <p
              className="jsx-430790654"
            >
              A table too:
            </p>
            <table>
              <thead>
                <tr>
                  <th>
                    Hello
                  </th>
                  <th>
                    world
                  </th>
                  <th>
                    table
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    some
                  </td>
                  <td>
                    values
                  </td>
                  <td>
                    here
                  </td>
                </tr>
                <tr>
                  <td>
                    some
                  </td>
                  <td>
                    values
                  </td>
                  <td>
                    here
                  </td>
                </tr>
                <tr>
                  <td>
                    some
                  </td>
                  <td>
                    values
                  </td>
                  <td>
                    here
                  </td>
                </tr>
                <tr>
                  <td>
                    some
                  </td>
                  <td>
                    values
                  </td>
                  <td>
                    here
                  </td>
                </tr>
              </tbody>
            </table>
            <p
              className="jsx-430790654"
            >
              Get listed
            </p>
            <ul>
              <li>
                Item One
              </li>
              <li>
                Item Two
              </li>
              <li>
                Item Three
              </li>
            </ul>
          </div>
          <div
            className="jsx-1163017566 content-placeholder"
          />
        </section>
      </div>
      <div
        className="jsx-1065517858 tag-chips"
      >
        <div
          className="jsx-1065517858"
          onClick={[Function]}
        >
          sir blog
        </div>
        <div
          className="jsx-1065517858"
          onClick={[Function]}
        >
          you're it
        </div>
        <div
          className="jsx-1065517858"
          onClick={[Function]}
        >
          No tiggy backs
        </div>
      </div>
      <div
        className="jsx-113928496 content-footer"
      >
        <section
          className="jsx-2442235184"
        >
          <h3
            className="jsx-2442235184"
          >
            Share the post
          </h3>
          <div
            className="jsx-2442235184"
          >
            <div
              className="jsx-2442235184"
            >
              <script
                className="jsx-2442235184"
                data-url="mocked-path"
                type="IN/Share"
              />
            </div>
            <div
              className="jsx-2442235184"
            >
              <a
                className="jsx-2442235184 twitter-share-button"
                data-show-count="false"
                data-text="My First Blog Post"
                href="https://twitter.com/share?ref_src=twsrc%5Etfw"
              >
                Tweet
              </a>
            </div>
          </div>
        </section>
      </div>
    </main>
    <footer
      className="jsx-27833366"
    >
      <section
        className="jsx-27833366"
      >
        <div
          className="jsx-27833366 footer-info"
        >
          <div
            className="jsx-27833366 footer-info-top"
          >
            <img
              alt="Amplience logo"
              className="jsx-2718936118 logo"
              src="/static/images/logo-100h.png"
              srcSet="/static/images/logo-50h.png 1x, /static/images/logo-100h.png 2x"
            />
            <div
              className="jsx-27833366 social-links"
            >
              <a
                className="jsx-3655793200"
                href="//twitter.com/amplience"
                rel="noreferrer"
                target="_new"
              >
                <img
                  alt="Twitter logo"
                  className="jsx-27833366"
                  src="/static/images/social/ic_twitter.svg"
                />
              </a>
              <a
                className="jsx-3655793200"
                href="//www.linkedin.com/company/amplience/"
                rel="noreferrer"
                target="_new"
              >
                <img
                  alt="Linkedin logo"
                  className="jsx-27833366"
                  src="/static/images/social/ic_linkedin.svg"
                />
              </a>
            </div>
          </div>
          <div
            className="jsx-27833366 copyright"
          >
            © Copyright Amplience 
            2019
          </div>
        </div>
        <div
          className="jsx-27833366 footer-links"
        >
          <section
            className="jsx-287747732 footer-links-section"
          >
            <h4
              className="jsx-287747732"
            >
              For Developers
            </h4>
            <ul
              className="jsx-287747732"
            >
              <li>
                <a
                  className="jsx-3655793200"
                  href="//github.com/amplience/"
                  rel="noreferrer"
                  target="_new"
                >
                  GitHub
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//stackoverflow.com/questions/tagged/amplience-dynamic-content"
                  rel="noreferrer"
                  target="_new"
                >
                  Stack Overflow
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//docs.amplience.net/integration/developergettingstarted.html"
                  rel="noreferrer"
                  target="_new"
                >
                  Dynamic Content docs
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//docs.amplience.net/contenthub.html"
                  rel="noreferrer"
                  target="_new"
                >
                  Content Hub docs
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//docs.amplience.net/integration/reference.html"
                  rel="noreferrer"
                  target="_new"
                >
                  API reference
                </a>
              </li>
            </ul>
          </section>
          <section
            className="jsx-287747732 footer-links-section"
          >
            <h4
              className="jsx-287747732"
            >
              Company
            </h4>
            <ul
              className="jsx-287747732"
            >
              <li>
                <a
                  className="jsx-3655793200"
                  href="//amplience.com"
                  rel="noreferrer"
                  target="_new"
                >
                  Website
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//amplience.com/careers"
                  rel="noreferrer"
                  target="_new"
                >
                  Careers
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//amplience.com/privacy-cookie-policy"
                  rel="noreferrer"
                  target="_new"
                >
                  Privacy policy
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//docs.amplience.net/releasenotes/readme.html"
                  rel="noreferrer"
                  target="_new"
                >
                  What\`s new
                </a>
              </li>
            </ul>
          </section>
        </div>
      </section>
    </footer>
    <div
      className="CookieConsent"
      style={
        Object {
          "alignItems": "center",
          "background": "rgba(255, 255, 255, 0.9)",
          "bottom": "0",
          "boxShadow": "20px 20px 20px 10px #000",
          "color": "#333",
          "flexDirection": "column",
          "position": "fixed",
          "textAlign": "center",
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <div
        className="cookie-banner-content"
        style={
          Object {
            "padding": "20px 0px 10px",
          }
        }
      >
        We use cookies to give you the best possible experience on our website. By continuing to browse this site, you give consent for cookies to be used. For more details please read our
         
        <a
          href="https://amplience.com/privacy-cookie-policy/"
          rel="noreferrer"
          target="_blank"
        >
          Cookie Policy
        </a>
        .
      </div>
      <div
        className=""
      >
        <button
          className="cookie-banner-btn"
          id="rcc-confirm-button"
          onClick={[Function]}
          style={
            Object {
              "background": "#fff",
              "border": "3px solid #CCC",
              "borderRadius": "5px",
              "color": "#333",
              "marginBottom": "10px",
            }
          }
        >
          Accept Cookies
        </button>
      </div>
    </div>
    <script
      async={true}
      src="https://www.google-analytics.com/analytics.js"
    />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "window.ga=window.ga||function(){(ga.q = ga.q || []).push(arguments)};ga.l=+new Date; ga('create', 'ga-tracking-id', 'auto'); ga('send', 'pageview');",
        }
      }
    />
  </div>
</div>
`;
